src/modules/course/schema/enrolledCourse.schema.ts

```diff
@@ -0,0 +1,41 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+import * as mongoose from 'mongoose';
+
+export type EnrolledCourseDocument = EnrolledCourse & Document;
+
+interface video {
+  num: number;
+  timestamp?: Date;
+}
+
+@Schema()
+export class EnrolledCourse {
+  @Prop({})
+  eId: mongoose.Schema.Types.ObjectId;
+
+  @Prop({ default: [false, false, false, false, false] })
+  videos_watched: boolean[];
+
+  @Prop({ default: [false, false, false, false, false] })
+  Assignments_done: boolean[];
+
+  @Prop({ default: [] })
+  currentVideo: video[];
+
+  @Prop({ default: [] })
+  doubts: string[];
+
+  @Prop({})
+  Course: mongoose.Schema.Types.ObjectId;
+}
+
+export const EnrolledCourseSchema = SchemaFactory.createForClass(
+  EnrolledCourse,
+);
+
+EnrolledCourseSchema.virtual('students', {
+  ref: 'User',
+  localField: 'eId',
```

Comment:
localField should be `_id` 
and foreign field should be `courseId`

Reply:
Oh yeah. i will change this one. I think that both should be _id because the field named courseId was removed